### PR TITLE
test(ProductiveCard): add avt test coverage

### LINKED ISSUE #6472: Evaluate ProductiveCard for WCAG 2.2 readiness
### Parent issue summary

> Product teams can confidently build WCAG 2.2 AA accessible products using Carbon for IBM Products.

- Increase transparency on what accessibility standards are currently met and tested.
- Increase accessibility coverage by introducing more automated tests.
- Evaluate, update, and add testing to ensure that we meet WCAG 2.2 (included in latest `accessibility-checker` rulesets).

### Acceptance criteria
- [ ] Test latest `accessibility-checker` rulesets on various states of productive card
- [ ] Add tests for disabled states
- [ ] Add tests for open and close states in overflow menus
- [ ] Review any additional states we may need to capture or add tests for
- [ ] Add keyboard navigation tests to AVT

### PR DESCRIPTION
**Pull request by devadula-nandan**
Closes #6472

adds accessibility tests for productive card

Since we are in v12 OverflowMenu. i noticed that passing aria-label prop to v12 OverflowMenu spreads it to the overflow menu container which is a div and is reported as violation
v12 OverflowMenu uses single `Label` prop to set tooltip text on menu button and aria-label for list container (ul)
which means we might need to deprecate either `iconDescription` or `overflowAriaLabel` as `Label` prop handles both with v12 OverflowMenu.

useful references
https://github.com/carbon-design-system/ibm-products/pull/5803
https://github.com/carbon-design-system/ibm-products/pull/6395

i have passed label as below, and awaiting for any review suggestions
```
label={iconDescription || overflowAriaLabel}
```

#### What did you change?
e2e/components/ProductiveCard/ProductiveCard-test.avt.e2e.js
packages/ibm-products/src/components/Card/Card.tsx

#### How did you test and verify your work?
yarn avt

### CODE REVIEW

File: e2e/components/ProductiveCard/ProductiveCard-test.avt.e2e.js [hunk 1]
```diff
@@ -23,4 +24,226 @@ test.describe('ProductiveCard @avt', () => {
...
+    await visitStory(page, {
+      component: 'ProductiveCard',
+      id: 'ibm-products-components-cards-productivecard--with-action-ghost-button',
+    });
+
+    await expect(page).toHaveNoACViolations('ProductiveCard @avt-disabled');
+
+    const buttons = page.locator(`button.${carbon.prefix}--btn`);
+    const disabledButton = buttons.nth(2);
+    expect(await disabledButton.getAttribute('disabled')).not.toBeNull();
```
Comment (elycheea) [inline e2e/components/ProductiveCard/ProductiveCard-test.avt.e2e.js:50]: I think we typically try to query by role or text over a specific query selector.

In this case, we could probably do 

```js
const disabledButton = page.getByRole('button', { name: 'Read more' });
```

Or some similar variation of that.

File: e2e/components/ProductiveCard/ProductiveCard-test.avt.e2e.js [hunk 2]
```diff
@@ -23,4 +24,226 @@ test.describe('ProductiveCard @avt', () => {
...
+  test('@avt-overflow-menu: validates overflow menu interactions', async ({
+    page,
+  }) => {
+    await visitStory(page, {
+      component: 'ProductiveCard',
+      id: 'ibm-products-components-cards-productivecard--with-overflow',
+    });
+
+    const menuButton = page.locator('button[aria-haspopup="true"]');
+    const menu = page.locator('ul[role="menu"]');
```
Comment (elycheea) [inline e2e/components/ProductiveCard/ProductiveCard-test.avt.e2e.js:78]: We can probably use the roles here as well. :)

https://playwright.dev/docs/locators#locate-by-role

**Review by elycheea — COMMENTED** [review]
Looks good overall, but I think there are a few places where we could also use role/text locators rather than by query selectors. :)

**Comment by devadula-nandan** [comment]
@elycheea fixed it

**Review by elycheea — APPROVED** [review]
Also agree it might be good to deprecate `iconDescription` or `overflowAriaLabel` if they are now redundant. If `iconDescription` overrides `overflowAriaLabel`, that may be preferable unless the latter is more widely used. 🤔

**Comment by devadula-nandan** [comment]
> Also agree it might be good to deprecate `iconDescription` or `overflowAriaLabel` if they are now redundant. If `iconDescription` overrides `overflowAriaLabel`, that may be preferable unless the latter is more widely used. 🤔

im pretty sure we can deprecate `iconDescription` as it has been introduced recently compared to `overflowAriaLabel`.
also changed the override order to prioritise `overflowAriaLabel`. and updated the pr accordingly.

although a more generic name `label` could be more convenient (just like we have in core. a common prop to pass to button tooltip and aria-label for v12 OverflowMenu) when compared to having `overflowAriaLabel` that also sets tooltip text.

any additional thoughts or suggestions?

**Review by elycheea — APPROVED** [review]

**Review by anamikaanu96 — CHANGES_REQUESTED** [review]
Hi @devadula-nandan can you add test for hover state.

**Review by elycheea — APPROVED** [review]

**Comment by devadula-nandan** [comment]
@anamikaanu96 done

**Review by anamikaanu96 — APPROVED** [review]
